src/services/user-service.js

```diff
@@ -50,6 +50,29 @@ class UserService {
     return this.userStore.save(user);
   }
 
+  async updateUserAccount({ userId, username, email }) {
+    logger.info('Updating account data for user with id %s', userId);
+
+    const otherExistingUser = await this.userStore.findOne({
+      $and: [
+        { _id: { $ne: userId } },
+        { $or: [{ username }, { email }] }
+      ]
+    });
+
+    if (otherExistingUser) {
```

Comment:
I think it would be cooler if it would check for both and collect the problems. I would hate to change my email and then realise after another request that I need to also change the user name :)